sklearn/cluster/k_means_.py

```diff
@@ -441,10 +442,19 @@ def _kmeans_single(X, n_clusters, x_squared_norms, max_iter=300,
             best_centers = centers.copy()
             best_inertia = inertia
 
-        if squared_norm(centers_old - centers) <= tol:
+        shift = squared_norm(centers_old - centers)
+        if shift <= tol:
             if verbose:
                 print("Converged at iteration %d" % i)
+
             break
+
+        if shift > 0:
```

Comment:
Sorry if I'm missing something but will this condition not always satisfy since shift is the squared_norm?